packages/c-select/src/c-select.tsx

```diff
@@ -0,0 +1,165 @@
+
+import { CSelectField, CSelectFieldProps } from "./c-select-field"
+
+interface RootProps extends Omit<HTMLChakraProps<"div">, "color"> {}
+
+interface SelectOptions extends FormControlOptions {
+  /**
+   * The border color when the select is focused. Use color keys in `theme.colors`
+   * @example
+   * focusBorderColor = "blue.500"
+   */
+  focusBorderColor?: string
+  /**
+   * The border color when the select is invalid. Use color keys in `theme.colors`
+   * @example
+   * errorBorderColor = "red.500"
+   */
+  errorBorderColor?: string
+  /**
+   * The placeholder for the select. We render an `<option/>` element that has
+   * empty value.
+   *
+   * ```jsx
+   * <option value="">{placeholder}</option>
+   * ```
+   */
+  placeholder?: string
+  /**
+   * The size (width and height) of the icon
+   */
+  iconSize?: string
+  /**
+   * The color of the icon
+   */
+  iconColor?: string
+}
+
+export interface CSelectProps
+  extends CSelectFieldProps,
+    ThemingProps<"Select">,
+    SelectOptions {
+  /**
+   * Props to forward to the root `div` element
+   */
+  rootProps?: RootProps
+}
+
+export const CSelect = defineComponent({
+  name: "CSelect",
+  props: {
+    color: String,
+    placeholder: String,
+    iconColor: String as PropType<CSelectProps["iconColor"]>,
+    isDisabled: Boolean as PropType<CSelectProps["isDisabled"]>,
+    isInvalid: Boolean as PropType<CSelectProps["isInvalid"]>,
+    focusBorderColor: SAO as PropType<CSelectProps["focusBorderColor"]>,
+    errorBorderColor: SAO as PropType<CSelectProps["errorBorderColor"]>,
+    ...vueThemingProps,
+  },
+  setup(props, { slots, attrs }) {
+    const styles = useMultiStyleConfig("Select", props)
+
+    const ownProps = computed(() => toRefs(reactive(omitThemingProps(props))))
+
+    const fieldProps = useFormControl(ownProps.value)
+    console.log(
+      "🚀 ~ file: c-select.tsx:104 ~ setup ~ fieldProps:",
+      fieldProps.value
+    )
+
+    const rootStyles: SystemStyleObject = {
+      width: "100%",
+      height: "fit-content",
+      position: "relative",
+      color: props.color,
+    }
+
+    const fieldStyles: SystemStyleObject = {
+      paddingEnd: "2rem",
+      ...styles.value.field,
+      _focus: {
+        zIndex: "unset",
+        ...(styles as any).value.field?.["_focus"],
+      },
+    }
+
+    const iconColor = computed(() => props.iconColor ?? props.color)
+
+    return () => (
+      <chakra.div __css={rootStyles}>
+        <CSelectField
+          placeholder={props.placeholder}
+          __css={fieldStyles}
+          {...fieldProps.value}
+          {...attrs}
+        >
+          {slots.default?.()}
+        </CSelectField>
+        <CSelectIcon
+          data-disabled={fieldProps.value.disabled}
+          {...fieldProps.value}
+          color={iconColor.value}
+          __css={styles.value.icon}
+        />
+      </chakra.div>
+    )
+  },
+})
+
+const CSelectIcon = defineComponent({
```

Comment:
Had issue setting up this component to allow a custom icon similar to the [react package implementation](https://github.com/chakra-ui/chakra-ui/blob/d6b6941c8e64f451313539d15f23fb0fb05ff129/packages/components/select/src/select.tsx#L165-L186). How should the approach be taken in vue since the custom icon is supposed to be passed in as a prop to `c-select`?

Reply:
Fixed.